src/NLog/Internal/PropertiesDictionary.cs

```diff
@@ -0,0 +1,744 @@
+        {
+            if (IsEmpty)
+                return System.Linq.Enumerable.Empty<KeyValuePair<object, object>>().GetEnumerator();
+            else
+                return new DictionaryEnumerator(this);
+        }
+
+        /// <inheritDoc/>
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            if (IsEmpty)
+                return ArrayHelper.Empty<KeyValuePair<object, object>>().GetEnumerator();
+            else
+                return new DictionaryEnumerator(this);
+        }
+
+        /// <inheritDoc/>
+        IDictionaryEnumerator IDictionary.GetEnumerator()
+        {
+            return new DictionaryEntryEnumerator(this);
+        }
+
+        /// <inheritDoc/>
+        public bool Remove(object key)
+        {
+            if (!IsEmpty)
+            {
+                return EventProperties.Remove(key);
+            }
+            return false;
+        }
+
+        /// <inheritDoc/>
+        public bool Remove(KeyValuePair<object, object> item)
+        {
+            if (!IsEmpty)
+            {
+                if (((IDictionary<object, PropertyValue>)EventProperties).Remove(new KeyValuePair<object, PropertyValue>(item.Key, new PropertyValue(item.Value, false))))
+                    return true;
+
+                if (((IDictionary<object, PropertyValue>)EventProperties).Remove(new KeyValuePair<object, PropertyValue>(item.Key, new PropertyValue(item.Value, true))))
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <inheritDoc/>
+        public bool TryGetValue(object key, out object value)
+        {
+            if (!IsEmpty)
+            {
+                PropertyValue valueItem;
+                if (EventProperties.TryGetValue(key, out valueItem))
+                {
+                    value = valueItem.Value;
+                    return true;
+                }
+            }
+
+            value = null;
+            return false;
+        }
+
+        /// <inheritDoc/>
+        bool IDictionary.Contains(object key)
+        {
+            return ContainsKey(key);
+        }
+
+        /// <inheritDoc/>
+        void IDictionary.Remove(object key)
+        {
+            Remove(key);
+        }
+
+        /// <inheritDoc/>
+        void ICollection.CopyTo(Array array, int arrayIndex)
+        {
+            if (array == null)
+                throw new ArgumentNullException("array");
+            if (arrayIndex < 0)
+                throw new ArgumentOutOfRangeException("arrayIndex");
+
+            if (!IsEmpty)
+            {
+                foreach (var propertyItem in this)
+                {
+                    array.SetValue(new DictionaryEntry(propertyItem.Key, propertyItem.Value), arrayIndex);
+                    ++arrayIndex;
+                }
+            }
+        }
+
+        void ResetMessageProperties()
+        {
+            if (_eventProperties != null && _messageProperties != null && _messageProperties.Count > 0)
```

Comment:
is this `if(IsEmpty)`?

Reply:
Will change it to IsEmpty (with some extra)

Reply:
Fixed